File: Queue/Queue.cs
namespace Struct.Queue;

using System;

class Queue
{
    private int _maxSize;
    private int _front;
    private int _rear;
    private int[] _queueArray;

    public Queue(int size)
    {
        _maxSize = size;
        _front = -1;
        _rear = -1;
        _queueArray = new int[_maxSize];
    }

    public void Enqueue(int value)
    {
        if (_rear == _maxSize - 1)
        {
            Console.WriteLine("Очередь полна. Невозможно добавить элемент.");
            return;
        }

        _queueArray[++_rear] = value;

        if (_front == -1)
        {
            _front = 0;
        }
    }

    public int Dequeue()
    {
        if (_front == -1 || _front > _rear)
        {
            Console.WriteLine("Очередь пуста. Невозможно извлечь элемент.");
            return -1;
        }

        int value = _queueArray[_front++];

        if (_front > _rear)
        {
            _front = -1;
            _rear = -1;
        }

        return value;
    }

    public int Peek()
    {
        if (_front == -1 || _front > _rear)
        {
            Console.WriteLine("Очередь пуста. Невозможно получить элемент.");
            return -1;
        }

        return _queueArray[_front];
    }

    public bool IsEmpty()
    {
        return (_front == -1 || _front > _rear);
    }

    public bool IsFull()
    {
        return (_rear == _maxSize - 1);
    }
}

File: Queue/Queue.py
class Queue:
    """
    Initializes the object by creating an empty queue.
    Parameters:
        self: The object itself.
    Return:
        None
    """
    def __init__(self):
        self.queue = []
    """
    Adds an item to the end of the queue.
    Parameters:
        item: The item to be added to the queue.
     Returns:
        None
    """
    def enqueue(self, item):
        self.queue.append(item)
    """
    Remove and return the first element in the queue.
    Parameters:
            None
    Returns:
            The first element in the queue if the queue is not empty, otherwise None.
    """
    def dequeue(self):
        if not self.is_empty():
            return self.queue.pop(0)
    """
    Check if the queue is empty.
    Returns:
        bool: True if the queue is empty, False otherwise.
    """
    def is_empty(self):
        return len(self.queue) == 0
    """
    Returns the size of the queue.
    :return: An integer representing the number of elements in the queue.
    """

    def size(self):
        return len(self.queue)